From 0630bf87b1a64d4492c2035e56d4cc03c2578f03 Mon Sep 17 00:00:00 2001
From: "Joshua M. Clulow" <jmc@joyent.com>
Date: Fri, 5 Jan 2018 00:59:12 +0000
Subject: [PATCH] MANTA-3477 create "manta-reshard" zone image Reviewed by:
 David Pacheco <dap@joyent.com> Approved by: David Pacheco <dap@joyent.com>

---
 config/services/reshard/service.json |  6 ++++++
 docs/man/man1/manta-adm.md           |  7 +++++--
 lib/layout.js                        | 10 ++++++++--
 lib/services.js                      |  6 ++++--
 man/man1/manta-adm.1                 |  7 +++++--
 test/tst.layout.js.out               |  8 ++++++++
 test/tst.services.js                 |  5 +++--
 7 files changed, 39 insertions(+), 10 deletions(-)
 create mode 100644 config/services/reshard/service.json

diff --git a/config/services/reshard/service.json b/config/services/reshard/service.json
new file mode 100644
index 0000000..fa21752
--- /dev/null
+++ b/config/services/reshard/service.json
@@ -0,0 +1,6 @@
+{
+	"params": {
+		"networks": [ "manta", "admin" ],
+		"ram": 512
+	}
+}
diff --git a/docs/man/man1/manta-adm.md b/docs/man/man1/manta-adm.md
index b1b9927..709fd56 100644
--- a/docs/man/man1/manta-adm.md
+++ b/docs/man/man1/manta-adm.md
@@ -1,4 +1,4 @@
-# MANTA-ADM 1 "2017" Manta "Manta Operator Commands"
+# MANTA-ADM 1 "2018" Manta "Manta Operator Commands"
 
 ## NAME
 
@@ -95,6 +95,9 @@ Services that are part of the Manta application include:
 **postgres**
   PostgreSQL databases used for storing object and job metadata
 
+**reshard**
+  Automated resharding system for Manta
+
 **storage**
   Stores actual object data
 
@@ -779,7 +782,7 @@ and `-y/--confirm` options described above.
 
 ## COPYRIGHT
 
-Copyright (c) 2017 Joyent Inc.
+Copyright (c) 2018, Joyent Inc.
 
 ## SEE ALSO
 
diff --git a/lib/layout.js b/lib/layout.js
index d50d342..a90ad70 100644
--- a/lib/layout.js
+++ b/lib/layout.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -85,7 +85,13 @@ var ML_SERVICES_EXACT = {
 	 * "propeller" is a testing component not intended for use in production
 	 * environments.
 	 */
-	'propeller': 0
+	'propeller': 0,
+
+	/*
+	 * "reshard" is an optional component for systems that are undergoing
+	 * resharding.
+	 */
+	'reshard': 0
 };
 
 /*
diff --git a/lib/services.js b/lib/services.js
index 64bf761..73eb1c2 100644
--- a/lib/services.js
+++ b/lib/services.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -50,7 +50,8 @@ var mSvcNames = [
     'ops',
     'madtom',
     'marlin-dashboard',
-    'marlin'
+    'marlin',
+    'reshard'
 ];
 
 /*
@@ -72,6 +73,7 @@ var mSvcConfigs = {
     'ops':		{ 'oneach': true,  'probes': true,  'sharded': false },
     'madtom':		{ 'oneach': true,  'probes': true,  'sharded': false },
     'marlin-dashboard':	{ 'oneach': true,  'probes': true,  'sharded': false },
+    'reshard':		{ 'oneach': true,  'probes': true,  'sharded': false },
     'marlin':		{ 'oneach': false, 'probes': false, 'sharded': false },
     'propeller':	{ 'oneach': true,  'probes': false, 'sharded': false }
 };
diff --git a/man/man1/manta-adm.1 b/man/man1/manta-adm.1
index 768304d..7377c21 100644
--- a/man/man1/manta-adm.1
+++ b/man/man1/manta-adm.1
@@ -1,4 +1,4 @@
-.TH MANTA\-ADM 1 "2017" Manta "Manta Operator Commands"
+.TH MANTA\-ADM 1 "2018" Manta "Manta Operator Commands"
 .SH NAME
 .PP
 manta\-adm \- administer a Manta deployment
@@ -91,6 +91,9 @@ Manages asynchronous operations like garbage collection, metering, and auditing
 \fBpostgres\fP
 PostgreSQL databases used for storing object and job metadata
 .TP
+\fBreshard\fP
+Automated resharding system for Manta
+.TP
 \fBstorage\fP
 Stores actual object data
 .TP
@@ -869,7 +872,7 @@ Generic failure.
 The command\-line options were not valid.
 .SH COPYRIGHT
 .PP
-Copyright (c) 2017 Joyent Inc.
+Copyright (c) 2018, Joyent Inc.
 .SH SEE ALSO
 .PP
 .BR json (1), 
diff --git a/test/tst.layout.js.out b/test/tst.layout.js.out
index d50b725..34d0a07 100644
--- a/test/tst.layout.js.out
+++ b/test/tst.layout.js.out
@@ -228,6 +228,7 @@ summary:
      madtom               -                1
      marlin-dashboard     -                1
      marlin               -               32
+     reshard              -                 
      postgres             1                3
      postgres             2                3
      postgres             3                3
@@ -608,6 +609,7 @@ summary:
      madtom               -                1
      marlin-dashboard     -                1
      marlin               -               16
+     reshard              -                 
      postgres             1                3
      moray                1                3
 --------------------------------------------------
@@ -717,6 +719,7 @@ summary:
      madtom               -                1
      marlin-dashboard     -                1
      marlin               -               16
+     reshard              -                 
      postgres             1                3
      postgres             2                3
      postgres             3                3
@@ -1031,6 +1034,7 @@ summary:
      madtom               -                1
      marlin-dashboard     -                1
      marlin               -               48
+     reshard              -                 
      postgres             1                3
      postgres             2                3
      moray                1                3
@@ -1513,6 +1517,7 @@ summary:
      madtom               -                1
      marlin-dashboard     -                1
      marlin               -               96
+     reshard              -                 
      postgres             1                3
      postgres             2                3
      postgres             3                3
@@ -1724,6 +1729,7 @@ summary:
      madtom               -                                 1                 
      marlin-dashboard     -                                                  1
      marlin               -               16               16               16
+     reshard              -                                                   
      postgres             1                1                1                1
      moray                1                1                1                1
 --------------------------------------------------
@@ -1948,6 +1954,7 @@ summary:
      madtom               -                                 1                 
      marlin-dashboard     -                                                  1
      marlin               -               16               16               16
+     reshard              -                                                   
      postgres             1                1                1                1
      postgres             2                1                1                1
      moray                1                1                1                1
@@ -2413,6 +2420,7 @@ summary:
      madtom               -                                 1                 
      marlin-dashboard     -                                                  1
      marlin               -               48               48               48
+     reshard              -                                                   
      postgres             1                1                1                1
      postgres             2                1                1                1
      postgres             3                1                1                1
diff --git a/test/tst.services.js b/test/tst.services.js
index bf2cf11..f86771f 100644
--- a/test/tst.services.js
+++ b/test/tst.services.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -38,7 +38,8 @@ var knownServices = [
     'madtom',
     'marlin-dashboard',
     'marlin',
-    'propeller'
+    'propeller',
+    'reshard'
 ];
 
 function main()
-- 
2.21.0

